src/main/java/com/capgemini/wsb/fitnesstracker/user/internal/UserRepository.java

```diff
@@ -2,23 +2,74 @@
 
 import com.capgemini.wsb.fitnesstracker.user.api.User;
 import org.springframework.data.jpa.repository.JpaRepository;
-import org.springframework.stereotype.Repository;
 
 import java.util.Objects;
 import java.util.Optional;
+import java.util.List;
+import java.util.stream.Collectors;
 
 interface UserRepository extends JpaRepository<User, Long> {
 
+
+    /**
+     * Save user
+     * @param user User
+     * @return User
+     */
+    default User saveUser(User user){
+        return save(user);
+    }
+
     /**
      * Query searching users by email address. It matches by exact match.
      *
      * @param email email of the user to search
      * @return {@link Optional} containing found user or {@link Optional#empty()} if none matched
      */
-    default Optional<User> findByEmail(String email) {
+    default List<User> findByEmail(String email) {
+        return findAll().stream()
+                .filter(user ->  user.getEmail().toLowerCase().startsWith(email.toLowerCase()))
```

Comment:
In requirements, it's specified that it should search by a phrase. The phrase can be in the beginning, in the middle or at the end. With current implementation it will work only with first case. If you decide to go with filtering it in java then this logic should be moved to service implementation.